leaktest_utils_test.go

```diff
@@ -0,0 +1,42 @@
+package leaktest
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"time"
+)
+
+func index() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	})
+}
+
+func startKeepAliveEnabledServer(ctx context.Context) {
+	router := http.NewServeMux()
+	router.Handle("/", index())
+
+	server := &http.Server{
+		Addr:         ":8091",
```

Comment:
Can you use `httptest.NewServer` here, so we're not forced to free up `:8091` on any machine running this?

Also kill the log statements :)

Reply:
Oooh cool, I'd not worked with the `httptest` stuff before that has made this much simpler!